### PR TITLE
Change trigger, agenda storage to store useriden instead of usernames

### PR DESCRIPTION
**Pull request by jnwatson**
Migrate v0 triggers to v1 triggers at startup.

Don't worry about migrating cron for now.

In a couple of provenance frames store the useriden instead of names.

Use string idens instead of binary ones

### CODE REVIEW

File: synapse/cortex.py [hunk 1]
```diff
@@ -193,38 +193,41 @@ def getCoreInfo(self):
...
         Lists all the triggers that the current user is authorized to access
         '''
-        username = None if self.user is None else self.user.name
-        return [(iden, trig) for (iden, trig) in self.cell.triggers.list()
-                if username is None or self.user.admin or trig.get('user') == username]
+        trigs = []
+        for (iden, trig) in self.cell.triggers.list():
+            if not (self.user.admin or trig['useriden'] == self.user.iden):
+                continue
+            trig['username'] = self.cell.auth.user(trig['useriden']).name
```
Comment (vEpiphyte) [inline synapse/cortex.py]: what happens if a user is deleted from the system that made a trigger? would this blow up and cause all triggers to be unlistable by a admin user?

File: synapse/cortex.py [hunk 2]
```diff
@@ -298,16 +300,21 @@ def _convert_reqdict(reqdict):
...
         Get information about all the cron jobs accessible to the current user
         '''
-        username = None if self.user is None else self.user.name
-        return [(iden, cron) for (iden, cron) in self.cell.agenda.list()
-                if username is None or self.user.admin or cron.get('user') == username]
+        crons = []
+        for iden, cron in self.cell.agenda.list():
+            if not (self.user.admin or cron['useriden'] == self.user.iden):
+                continue
+            cron['username'] = self.cell.auth.user(cron['useriden']).name
```
Comment (vEpiphyte) [inline synapse/cortex.py]: same problem as noted above - what happens here if the user is gone from the hive?

File: synapse/lib/agenda.py [hunk 3]
```diff
@@ -584,22 +584,19 @@ def list(self):
...
         '''
-        if appt.username is None or self.core.auth is None:
-            user = None
-        else:
-            user = self.core.auth.getUserByName(appt.username)
-            if user is None:
-                logger.warning('Unknown username %s in stored appointment', appt.username)
-                return
-        await self.core.boss.execute(self._runJob(user, appt), f'Agenda {s_common.ehex(appt.iden)}', user)
+        user = self.core.auth.user(appt.useriden)
```
Comment (vEpiphyte) [inline synapse/lib/agenda.py]: same problem of a missing user :(

File: synapse/lib/agenda.py [hunk 4]
```diff
@@ -584,22 +584,19 @@ def list(self):
...
+        if user is None:
+            logger.warning('Unknown user %s in stored appointment', appt.useriden)
```
Comment (vEpiphyte) [inline synapse/lib/agenda.py]: should we mark the appointment here as failed?

File: synapse/lib/cell.py [hunk 5]
```diff
@@ -69,8 +70,7 @@ def getCellUser(self):
         retn = []
 
         admin = False
-        if self.user is not None:
-            admin = self.user.admin
+        admin = self.user.admin
```
Comment (vEpiphyte) [inline synapse/lib/cell.py]: can remove the previous `admin = False`

File: synapse/lib/cell.py [hunk 6]
```diff
@@ -81,8 +81,7 @@ def getCellUser(self):
     async def kill(self, iden):
 
         admin = False
-        if self.user is not None:
-            admin = self.user.admin
+        admin = self.user.admin
```
Comment (vEpiphyte) [inline synapse/lib/cell.py]: can remove the previous admin = False

File: synapse/lib/trigger.py [hunk 7]
```diff
@@ -169,21 +166,52 @@ def __init__(self, core):
...
+            # In v0, stored user was username, in >0 user is useriden
+            user = self.core.auth.getUserByName(user).iden
+            if user is None:
+                logger.warning('Unrecognized username in stored trigger rule %r', iden)
+                continue
+
+            ruledict['ver'] = 1
+            ruledict['useriden'] = user
+            newiden = s_common.ehex(iden)
+            self.core.slab.pop(iden)
```
Comment (vEpiphyte) [inline synapse/lib/trigger.py]: don't we need a db argument here?

File: synapse/cortex.py (hunk 1 above)
Comment (jnwatson) [reply synapse/cortex.py]: Good point. Fixed.

File: synapse/cortex.py (hunk 2 above)
Comment (jnwatson) [reply synapse/cortex.py]: Fixed same way.

File: synapse/lib/agenda.py (hunk 4 above)
Comment (jnwatson) [reply synapse/lib/agenda.py]: ok

File: synapse/lib/cell.py (hunk 5 above)
Comment (jnwatson) [reply synapse/lib/cell.py]: ok

File: synapse/lib/trigger.py (hunk 7 above)
Comment (jnwatson) [reply synapse/lib/trigger.py]: Yes. Good catch.

File: synapse/lib/cell.py (hunk 6 above)
Comment (jnwatson) [reply synapse/lib/cell.py]: ok

**Pull request merged**